project/index.mjs

```diff
@@ -204,8 +204,135 @@ router.route("/:id/remove-member").post(auth0Middleware(), async (req, res) => {
+
+    if (!accessInfo.hasAccess) return respondWithError(res, 403, accessInfo.message)
+
+    const groupId = projectObj.data.group
+    const group = new Group(groupId)
+    await group.addMemberRoles(collaboratorId, roles)
+    await group.update()
+    res.status(200).json({ message: `Roles added to member ${collaboratorId}.` })
+  } catch (error) {
+    res.status(error.status || 500).json({ message: error.message || "Error adding roles to member." })
+  }
 })
 
+
+// Change a member's Role(s): Replace roles with new ones
+router.route("/:projectId/collaborator/:collaboratorId/setRoles").put(auth0Middleware(), async (req, res) => {
+  const { projectId, collaboratorId } = req.params
+  const roles = req.body.roles ?? req.body
+  const user = req.user
+  if (!user) {
+    return respondWithError(res, 401, "Unauthenticated request")
+  }
+
+  if (!roles) {
+    return respondWithError(res, 400, "Provide role(s) to update")
+  }
+
+
+  try {
+    const projectObj = await new Project(projectId)
+    const accessInfo = await projectObj.checkUserAccess(user._id, ACTIONS.UPDATE, SCOPES.ALL, ENTITIES.MEMBER)
+
+    if (!accessInfo.hasAccess) return respondWithError(res, 403, accessInfo.message)
+
+    const groupId = projectObj.data.group
+    const group = new Group(groupId)
+    group.setMemberRoles(collaboratorId, roles)
+
+    res.status(200).json({ message: `Roles [${roles}] updated for member ${collaboratorId}.` })
+  } catch (error) {
+    res.status(error.status || 500).json({ message: error.message || "Error updating member roles." })
+  }
+})
+
+
+// Remove a Role from Member
+router.route("/:projectId/collaborator/:collaboratorId/removeRoles").post(auth0Middleware(), async (req, res) => {
+  const { projectId, collaboratorId } = req.params
+  const roles = req.body.roles ?? req.body
+  const user = req.user
+  if (!user) {
+    return respondWithError(res, 401, "Unauthenticated request")
+  }
+  if (!roles) {
+    return respondWithError(res, 400, "Provide role(s) to remove")
+  }
+  if (roles.includes("OWNER")) {
+    return respondWithError(res, 400, "The OWNER role cannot be removed.")
+  }
+  try {
+    const projectObj = await new Project(projectId)
+    const accessInfo = await projectObj.checkUserAccess(user._id, ACTIONS.DELETE, SCOPES.ALL, ENTITIES.MEMBER)
+
+    if (!accessInfo.hasAccess) return respondWithError(res, 403, accessInfo.message)
+
+    const groupId = projectObj.data.group
+    const group = new Group(groupId)
+    await group.removeMemberRoles(collaboratorId, roles)
+
+    res.status(204).send()
+
+  } catch (error) {
+    res.status(error.status || 500).json({ message: error.message || "Error removing roles from member." })
+  }
+})
+
+
+// Switch project owner
+
+router.route("/:projectId/switch/owner").post(auth0Middleware(), async (req, res) => {
+  const { projectId } = req.params
+  const { newOwnerId } = req.body
```

Comment:
How can I provide this new owner id?  It doesn't attempt to read from it as JSON and when I do content-type: "text/html" and provide a string ID I still get the "Provide the ID of the new owner." error.  I couldn't find a way to manually test this endpoint.

Reply:
This is what the endpoint expects as a body
```
 {
   "newOwnerId":"xyz"
 } 
```